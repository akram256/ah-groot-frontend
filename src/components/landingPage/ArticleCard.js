import React, { Component } from 'react';
import moment from 'moment';
import sanitizeHtml from 'sanitize-html';
import { withRouter } from 'react-router-dom';


export class ArticleCard extends Component {
  render() {
    const {
      title,
      description,
      body,
      reading_time,
      updated_at,
      author,
      slug,
    } = this.props;
    const newdate = new Date(updated_at).toString();
    const timeAgo = moment(`${newdate}`).fromNow();

    return (
      <div className="col s6">
        <div className="card grey lighten-5">
          <div className="card-content black-text"  slug={slug}
                      onClick={(event) => {
                        const slug = event.currentTarget.getAttribute('slug');
                        this.props.history.push(`/article/${slug}/view`);
                      }}>
            <span className="card-title truncate">{title}</span>
            <span className="card-description truncate">{description}</span>
            <div className="truncate">
              {sanitizeHtml(body, { allowedTags: [], allowedAttributes: {} })}
            </div>
          </div>
          <div className="card-action all-article-details">
            <div>
              published by {author} {timeAgo}
            </div>
            <div>{reading_time}</div>
            <i className="material-icons">star</i>
          </div>
        </div>
      </div>
    );
  }
}

export default withRouter(ArticleCard);
