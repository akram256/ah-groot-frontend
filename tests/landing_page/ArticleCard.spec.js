import React from 'react';
import { shallow } from 'enzyme';

import ArticleCard, { ArticleCard as App} from '../../src/components/landingPage/ArticleCard';

import { BrowserRouter as Router } from 'react-router-dom';
describe('ArticleCard', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<ArticleCard />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should respond to click events', () => {
    const wrapper = shallow(<App history={{ push: jest.fn()}}/>);
    const event = {
      currentTarget: {
        getAttribute:  jest.fn(),
      }
    };
    wrapper.find('.card-content.black-text').simulate('click', event);
    expect(wrapper).toMatchSnapshot();
  });
});
