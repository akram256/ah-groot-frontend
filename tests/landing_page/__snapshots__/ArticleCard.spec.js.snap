// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard should render without crashing 1`] = `ShallowWrapper {}`;

exports[`ArticleCard should respond to click events 1`] = `ShallowWrapper {}`;
